--- a/png.h.orig	Fri Jul 17 05:06:01 2015
+++ b/png.h	Fri Jul 17 05:09:11 2015
@@ -415,6 +415,10 @@
 #ifndef PNG_H
 #define PNG_H
 
+#pragma lib "/$M/lib/ape/libpng.a"
+/* need to pull in libz.a too */
+#include <zlib.h>
+
 /* This is not the place to learn how to use libpng. The file libpng-manual.txt
  * describes how to use libpng, and the file example.c summarizes it
  * with some code on which to build.  This file is useful for looking
