diff -u sbase-master/libutil.orig/cp.c sbase-master/libutil/cp.c
--- sbase-master/libutil.orig/cp.c	Sun May 17 09:57:01 2015
+++ sbase-master/libutil/cp.c	Sun May 17 10:00:56 2015
@@ -154,9 +154,11 @@
 	if (cp_aflag || cp_pflag) {
 		/* timestamp and owner*/
 		if (!S_ISLNK(st.st_mode)) {
+/*
 			times[0] = st.st_atim;
 			times[1] = st.st_mtim;
 			utimensat(AT_FDCWD, s2, times, 0);
+*/
 
 			if (chown(s2, st.st_uid, st.st_gid) < 0) {
 				weprintf("chown %s:", s2);
diff -u sbase-master/libutil.orig/ealloc.c sbase-master/libutil/ealloc.c
--- sbase-master/libutil.orig/ealloc.c	Sun May 17 09:57:01 2015
+++ sbase-master/libutil/ealloc.c	Sun May 17 10:00:32 2015
@@ -81,7 +81,7 @@
 {
 	char *p;
 
-	p = strndup(s, n);
+	p = strdup(s);
 	if (!p)
 		enprintf(status, "strndup: out of memory\n");
 	return p;
