--- make-4.1/function.orig	Wed Jun 10 19:54:02 2015
+++ make-4.1/function.c	Wed Jun 10 19:56:21 2015
@@ -1,6 +1,6 @@
 /* Builtin function expansion for GNU Make.
 Copyright (C) 1988-2014 Free Software Foundation, Inc.
-This file is part of GNU Make.
+This file is part of GNU Make. alloca
 
 GNU Make is free software; you can redistribute it and/or modify it under the
 terms of the GNU General Public License as published by the Free Software
@@ -37,8 +37,8 @@
     unsigned char len;
     unsigned char minimum_args;
     unsigned char maximum_args;
-    unsigned char expand_args:1;
-    unsigned char alloc_fn:1;
+    unsigned char expand_args;
+    unsigned char alloc_fn;
   };
 
 static unsigned long
@@ -956,7 +956,7 @@
   pattail = &pathead;
   while ((p = find_next_token (&pat_iterator, &len)) != 0)
     {
-      struct a_pattern *pat = alloca (sizeof (struct a_pattern));
+      struct a_pattern *pat = (struct a_pattern *) alloca (sizeof (struct a_pattern));
 
       *pattail = pat;
       pattail = &pat->next;
@@ -980,7 +980,7 @@
   wordtail = &wordhead;
   while ((p = find_next_token (&word_iterator, &len)) != 0)
     {
-      struct a_word *word = alloca (sizeof (struct a_word));
+      struct a_word *word = (struct a_word *) alloca (sizeof (struct a_word));
 
       *wordtail = word;
       wordtail = &word->next;
@@ -2402,7 +2402,7 @@
   *stringp = end;
 
   /* Get some memory to store the arg pointers.  */
-  argvp = argv = alloca (sizeof (char *) * (nargs + 2));
+  argvp = argv = (char **) alloca (sizeof (char *) * (nargs + 2));
 
   /* Chop the string into arguments, then a nul.  As soon as we hit
      MAXIMUM_ARGS (if it's >0) assume the rest of the string is part of the
