--- /dev/null	Mon Sep 10 15:20:26 2018
+++ b/cmd/commands_ssh_plan9.go	Wed Sep 12 16:44:43 2018
@@ -0,0 +1,28 @@
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+)
+
+func connectSSH(user, host, key string, port int) {
+	args := []string{"-u", user}
+
+	if port == 22 {
+		args = append(args, host)
+	} else {
+		args = append(args, fmt.Sprintf("tcp!%s!%d", host, port))
+	}
+
+	cmd := exec.Command("ssh", args...)
+
+	cmd.Stdin = os.Stdin
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+
+	if err := cmd.Run(); err != nil {
+		log.Fatal(err)
+	}
+}
