--- a/linit.c	Mon Jun  1 04:42:02 2015
+++ b/linit.c	Mon Jun  1 05:23:59 2015
@@ -23,6 +23,10 @@
   {LUA_STRLIBNAME, luaopen_string},
   {LUA_MATHLIBNAME, luaopen_math},
   {LUA_DBLIBNAME, luaopen_debug},
+  {LUA_BITLIBNAME, luaopen_bit},
+  {LUA_CJSONLIBNAME, luaopen_cjson},
+  {LUA_CMSGPACKLIBNAME, luaopen_cmsgpack},
+  {LUA_STRUCTLIBNAME, luaopen_struct},
   {NULL, NULL}
 };
 
--- a/lualib.h	Mon Jun  1 04:42:02 2015
+++ b/lualib.h	Mon Jun  1 05:23:00 2015
@@ -39,11 +39,20 @@
 #define LUA_LOADLIBNAME	"package"
 LUALIB_API int (luaopen_package) (lua_State *L);
 
+#define LUA_BITLIBNAME "bit"
+LUALIB_API int (luaopen_bit) (lua_State *L);
 
-/* open all previous libraries */
-LUALIB_API void (luaL_openlibs) (lua_State *L); 
+#define LUA_CJSONLIBNAME "cjson"
+LUALIB_API int (luaopen_cjson) (lua_State *L);
+
+#define LUA_CMSGPACKLIBNAME "cmsgpack"
+LUALIB_API int (luaopen_cmsgpack) (lua_State *L);
 
+#define LUA_STRUCTLIBNAME "struct"
+LUALIB_API int (luaopen_struct) (lua_State *L);
 
+/* open all previous libraries */
+LUALIB_API void (luaL_openlibs) (lua_State *L); 
 
 #ifndef lua_assert
 #define lua_assert(x)	((void)0)
